### ISSUE TITLE
iOS BundleResource Missing / Not Working

### ISSUE THREAD

**Issue by PureWeen** [xamarin/xamarin-macios#15347]
Moved from https://github.com/dotnet/maui/issues/8301

### Description

For firebase services the the "GoogleService-Info.plist" file build action has to be set as "BundleResource". The build action is not available in the visual studio selecten and also does not work when setting in the project file:

```
    <ItemGroup>
      <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" />
    </ItemGroup>
```

The following errror occurs when "Firebase.Core.App.Configure()" is called:
`ObjCRuntime.ObjCException: 'Objective-C exception thrown.  Name: com.firebase.core Reason: FirebaseApp.configure() could not find a valid GoogleService-Info.plist in your project. Please download one from https://console.firebase.google.com/.
Native stack trace:`

### Steps to Reproduce

1. dotnet new maui
2. Install nuget Xamarin.Firebase.iOS.CloudMessaging
3. Add GoogleService-Info.plist file to Platforms/iOS
4. Try to set build action
5. Call Firebase.Core.App.Configure(); in AppDelegate 

```
    public override bool FinishedLaunching(UIApplication application, NSDictionary launchOptions)
    {
        Firebase.Core.App.Configure();
        return base.FinishedLaunching(application, launchOptions);
    }
```

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

All

### Did you find any workaround?

Yes the configuration can be set in code

```
Firebase.Core.App.Configure(new Options("1:7..4:ios:8..7", "7..4")
{
	ApiKey = "A.._W.._8.._g",
	ClientId = "...apps.googleusercontent.com",
	BundleId = "<<BUNDLE-ID>>",
	ProjectId = "<<ProjectId>>"
});
```

**Comment by PureWeen** [comment]
> Have you looked inside the generated bundle to see if `GoogleService-Info.plist` is being copied anywhere?
> 
> Please attach a build log, ideally a binary build log, so we can look into what's going on.

@epsmae

**Comment by epsmae** [comment]
@PureWeen sorry for my late reply:
1. dotnet new maui
2. Added Platforms\iOS\GoogleService-Info.plist file
3. Add the following to csproj

```
    <ItemGroup>
      <BundleResource  Include="Platforms\iOS\GoogleService-Info.plist" />
    </ItemGroup>
```

Following the build log hope it helps (Note publish failed due to invalid publish profile)
[Verbose.log](https://github.com/xamarin/xamarin-macios/files/9061345/Verbose.log)

**Comment by chamons** [comment]
It appears that file is being added to your bundle in `Platforms/iOS/GoogleService-Info.plist` 

```
 CopyingFile - Path: /Users/epsag/Library/Caches/Xamarin/mtbs/builds/AppBundleSample/14c0993197954ee38641618911ebdf8b1574d67cb49cb38b0dbf79d6e997d01a/bin/Debug/net6.0-ios/ios-arm64/device-builds/iphone13.2-15.5/AppBundleSample.app/Platforms/iOS/GoogleService-Info.plist
```

I assume that it needs to be in the root of your bundle?

**Comment by epsmae** [comment]
@chamons thanks, this works indeed. However I do not believe that any developer stumbles across this document.
My assumption is/was that the platform folder is already handled as root and i was not aware of the link property.

So I believe the bug or missing features is then that the build action is not shown in a maui project but you can select it in an net6-ios project.

**Comment by chamons** [comment]
I agree the behavior is less than ideal from a MAUI context, I'm going to transfer this bug over there for consideration.

**Comment by chamons** [comment]
This issue was moved to dotnet/maui#8612
